src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -70,6 +70,14 @@ abstract class UnCurry extends InfoTransform
     private val noApply           = mutable.HashSet[Tree]()
     private val newMembers        = mutable.Map[Symbol, mutable.Buffer[Tree]]()
 
+    private lazy val forceSpecializationInfoTransformOfFunctionN: Unit = {
+      if (currentRun.specializePhase != NoPhase) {
```

Comment:
If it's not clear why it's needed, should we remove it? This kind of code is hard to remove later on because of the fear hard to predict breakage.